light-client/src/components/scheduler.rs

```diff
@@ -1,14 +1,25 @@
 use crate::prelude::*;
```

Comment:
Maybe in the preamble provide reference to the english spec  such that the invariants can be looked up by the reader.

Reply:
@ebuchman suggested getting rid of the prelude entirely for better discoverability so that's perhaps not the right place to put those references. The crate-level documentation in `lib.rs` might be a better fit.